tock/tock/templates/uaa_client/login_error.html

```diff
@@ -1,22 +1,20 @@
 {% extends "base.html" %}
-{% block content %}
 
-<h2>cloud.gov Login Error</h2>
+{% block content %}
 
-{% if error_code == "missing_state" or
-      error_code == "missing_session_state" or
-      error_code == "invalid_state" %}
+  <h2>cloud.gov Login Error</h2>
 
-<p>Bad session state</p>
+  {% if "_state" in error_code %}
 
-{% elif error_code == "missing_code" %}
+      <p>Bad session state</p>
 
-<p>Missing session code</p>
+  {% elif error_code == "missing_code" %}
 
-{% elif error_code == "authenticate_failed" %}
+    <p>Missing session code</p>
 
-<p>Authentication failed</p>
+  {% else %}
```

Comment:
Hey @tbaxter-18f I'm not sure why you're removing the explicit check for `error_code == "authenticate_failed"` here? I like the approach of the substring match 👍 and in general having a catchall like an empty `else` is good in the case the underlying `uaa_client` code changes. 

<details>
<summary>
<a href="https://github.com/18F/cg-django-uaa/blob/maser/uaa_client/views.py#L39-L55">Here's</a> where I'm getting these values from.
</summary>

```py
    if state is None:
        return login_error(request, 'missing_state')


    if expected_state is None:
        return login_error(request, 'missing_session_state')


    if state != expected_state:
        return login_error(request, 'invalid_state')


    if code is None:
        return login_error(request, 'missing_code')


    user = django.contrib.auth.authenticate(uaa_oauth2_code=code,
                                            request=request)


    if user is None:
        return login_error(request, 'authenticate_failed')
```
</details>

Reply:
I started to, and the more I thought about it, the more I though it was unnecessary bloat. My logic was that we've already captured the expected failure states. if we get to here, it's either a straight-up generic authentication failed, or something really weird we didn't foresee. So either way, we can treat is a generic error, provide the same generic message, and output what the code was for further investigation.  I'm not strongly opposed to having a separate authenticate_failed block, but I don't think it's necessary or even useful.

Reply:
hmm yeah. I think that's fair! let's stick to that logic then. Thinking about it more, my concern is with exposing the `error_code` variable to the user. I think it'd be better to just log it rather than expose it. That way we don't give folks who encounter it in the wild any insight into what caused their issue, but people with log access can determine it pretty quickly. Maybe we should just update the [`uaa_client` code here](https://github.com/18F/cg-django-uaa/blob/maser/uaa_client/views.py#L15) to output a generic log message. Or in the [new functions I added](https://github.com/18F/tock/blob/master/tock/tock/views.py#L25-L70) to `tock/views.py` maybe but I don't think the `uaa_client` is hitting those functions.

I can approve this based on this explanation, but what do you think of just sending that `error_code` to logs rather than the HTML view the user sees?